keystore/v2/keystore/keyStore.go

```diff
@@ -163,3 +224,39 @@ func (s *ServerKeyStore) DescribeKeyRing(path string) (*keystore.KeyDescription,
 // KeyStoreV2 currently does not support key caching so there is nothing to reset.
 func (s *ServerKeyStore) Reset() {
 }
+
+func (s *ServerKeyStore) listHistoricalRings(path string, purpose keystore.KeyPurpose, clientID []byte) ([]keystore.KeyDescription, error) {
+	ring, err := s.OpenKeyRing(path)
+	if err != nil {
+		log.WithError(err).Debug("Failed to open audit log key ring")
+		return nil, err
+	}
+
+	keys, err := ring.AllKeys()
+	if err != nil {
+		log.WithError(err).Debug("Failed to read all key ids")
+		return nil, err
+	}
+
+	if len(keys) == 0 {
+		return []keystore.KeyDescription{}, nil
+	}
+
+	result := make([]keystore.KeyDescription, 0, len(keys)-1)
+	for i := len(keys) - 1; i > 0; i-- {
```

Comment:
this complicated iteration pushed to think that we should test an order of result lists for rotated keys to be sure that we return and list them in the same order for v1 and v2 keystores